client-apis/streaming.md

```diff
@@ -0,0 +1,111 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Streaming
+
+## Version
+
+0.1.0
+
+## Changelog
+
+- 0.1.0
+
+  - [Clarify Streaming Encrypt and Decrypt](../changes/2020-07-06_clarify-streaming-encrypt-decrypt/change.md)
+
+## Overview
+
+The AWS Encryption SDK MAY provide APIs that enable streamed [encryption](encrypt.md)
+and [decryption](decrypt.md).
+Streaming is a framework for making bytes available to be processed
+by an operation sequentially and over time,
+and for outputting the result of that processing
+sequentially and over time.
+
+If an implementation requires holding the entire input in memory in order to perform the operation,
+that implementation SHOULD NOT provide an API that allows the caller to stream the operation.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC2119](https://tools.ietf.org/html/rfc2119).
+
+### Available Bytes
```

Comment:
This term might be a little overloaded - your description is right in that just because bytes COULD be read doesn't mean they are INTENDED to be. I ended up using "consumable" in my comments to mean the former instead.

Reply:
Should the header be "Consumable Bytes" too then?